airflow/providers/databricks/hooks/databricks_sql.py

```diff
@@ -167,7 +195,7 @@ def run(
         handler: Callable[[Any], T] = ...,
         split_statements: bool = ...,
         return_last: bool = ...,
-    ) -> T | list[T]:
+    ) -> tuple | list[tuple] | list[list[tuple] | tuple] | None:
```

Comment:
Mypy was not happy with `-> tuple | list[tuple] | list[list[tuple] | tuple]` as output. I got *Overloaded function signatures 1 and 2 overlap with incompatible return types*.

[Following mypy docs](https://mypy.readthedocs.io/en/stable/more_types.html#type-checking-the-variants), I added a `None` for it to be stop complaining. But that's not correct, this method cannot return None when a handler is provided...

Implementing definitive typing is out of the scope of this PR, thus I let that for #36224.